jax/_src/lax/parallel.py

```diff
@@ -288,13 +296,19 @@ def all_to_all(x, axis_name, split_axis, concat_axis):
     where ``axis_size`` is the size of the mapped axis named ``axis_name`` in
     the input ``x``, i.e. ``axis_size = lax.psum(1, axis_name)``.
   """
+  if axis_index_groups and split_axis != concat_axis:
+    raise ValueError("all_to_all only supports axis_index_groups if "
+                     "split_axis = concat_axis")
```

Comment:
It is 100% ok to have this for now, but I'm curious as to why did you have to add this restriction?

Reply:
It was actually a misconception from my side, it works perfectly with split_axis != concat_axis. I've removed this check and updated the test case.